pkg/image/registry/imagesignature/rest.go

```diff
@@ -0,0 +1,98 @@
+package imagesignature
+
+import (
+	"errors"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kapierrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/rest"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	"github.com/openshift/origin/pkg/image/api/validation"
+	"github.com/openshift/origin/pkg/image/registry/image"
+)
+
+// REST implements the RESTStorage interface for ImageSignature
+type REST struct {
+	imageRegistry image.Registry
```

Comment:
> Yes, use a client. It's an infrequent API, it's admin only, and you're only updating the image itself, not status, right?

The rest doesn't touch any image stream. Just the image which has no status.
Using the client sounds good to me. I'll rewrite.